src/inet/UDPEndPoint.cpp

```diff
@@ -649,6 +649,10 @@ INET_ERROR UDPEndPoint::SendMsg(const IPPacketInfo * pktInfo, System::PacketBuff
     SuccessOrExit(res);
 
     res = IPEndPointBasis::SendMsg(pktInfo, std::move(msg), sendFlags);
+
+    // Wait for ability to write on this endpoint.
```

Comment:
In what cases do we expect the socket not to be ready for write?  In some applications, you might set a UDP write socket to block with the assumption that usually the buffers won't be full and therefore most writes actually won't block.

Is it correct though that we aren't operating this way?

Reply:
It sounds like mbedOS has this at least, but also - doesn't seem harmful? All our interaction tests seem to pass. Is there something here that's breaking?

Reply:
No, just trying to understand.

Thank you.

Reply:
In Matter, all the `SendMsg` calls are made with the `sendFlags` parameters set to `0`. It always block and previous code is ok in such case - even with Mbed OS. 

However, it doesn't work at all if you want to use an UDP endpoint as an _asynchronous_ client because the socket wasn't added to the list of FD monitored by select when `SendMsg` is called. We experienced this in Mbed OS while developing our POSIX adaptation layer (`HandlePendingIO` was slightly modified to test the sendmsg part).

This change ensure the FD is added to the list of FD monitored by select and makes it possible to implement such case. Of course it would require updating `UDPEndpoint::HandlePendingIO` to properly handle the case of a pending write.

Reply:
Note that on Mbed OS waking up the `select` thread has the side effect of giving more CPU time for the networking threads. That helps for some drivers that otherwise silently drop UDP packets queued by LwIP during the _blocking_ `sendmsg` as depending on the underlying driver it might only be blocking at the LwIP level due to EMAC controller implementation limitations.